tests/quantization/bnb/test_4bit_serialization.py

```diff
@@ -0,0 +1,189 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a clone of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import tempfile
+import unittest
+
+from transformers import (
+    AutoConfig,
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    BitsAndBytesConfig,
+)
+from transformers.testing_utils import (
+    is_bitsandbytes_available,
+    require_accelerate,
+    require_bitsandbytes,
+    require_torch,
+    require_torch_gpu,
+)
+
+
+if is_bitsandbytes_available():
+    import bitsandbytes as bnb
+    import torch
+
+
+def get_some_linear_layer(model):
+    if model.config.model_type == "gpt2":
+        return model.transformer.h[0].mlp.c_fc
+    elif model.config.model_type == "opt":
+        return model.model.decoder.layers[0].fc1
+    else:
+        return model.transformer.h[0].mlp.dense_4h_to_h
+
+
+@require_bitsandbytes
+@require_accelerate
+@require_torch
+@require_torch_gpu
+# @slow
+class BaseTest(unittest.TestCase):
```

Comment:
Can you move this test inside `tes4bit.py`?

Reply:
done